site/ja/tfx/index.md

```diff
@@ -0,0 +1,585 @@
+このコンポーネントはデータの中に異常値や欠損値、誤ったデータ型が存在しないか検査します。
+これらのコンポーネントは [TensorFlow Data Validation](tfdv.md) の能力を活用しています。
+
+[TensorFlow Data Validation (TFDV)](tfdv.md) はデータセットの探索、可視化、クリーニングを
+行う際に役に立つツールです。
+TFDV はデータを確認してデータの型、カテゴリ、値域を推定し、その後、自動的に異常値や欠損値を特定するのを手助けします。
+パイプラインのコンポーネントの処理が完了したあと、 [MLMD](mlmd.md) からメタデータを読み込み、
+データを分析するために TFDV の可視化ツールを Jupyter ノートブック上で利用できます。
+最初にモデルがデプロイされたあとに、TFDV をデプロイされたモデルへの推論リクエストに含まれる
+新たなデータを監視し、異常値やドリフトの検出を行うために利用できます。
+これは時系列データでトレンドや季節性があり、時間の経過にしたがって変化するものに対してはとくに有効で、
+データの問題やモデルを新しいデータで再学習させる必要があるときに、通知を行う手助けになります。
+
+### データの可視化
+
+パイプラインのうち、TFDV を利用するコンポーネント (典型的には StatisticsGen, SchemaGen, ExampleValidator) では、データを処理した結果を Jupyter notebook で可視化できます。
+データがモデルやアプリケーションに最適化されるまで何度も実行して結果を比較し、調整を行うことも可能です。
+
+これらのコンポーネントの出力結果を [**ML Metadata (MLMD)**](mlmd.md) にクエリを発行して保存したあと notebook で可視化するために、 TFDV の可視化をサポートする API を利用できます。
+これには [tfdv.load_statistics()](`tfdv.load_statistics`) や [tfdv.visualize_statistics()](`tfdv.visualize_statistics`) が含まれます。
+これらの可視化を利用することで、データセットの特徴についてよりよい理解を得ることや、もし必要な場合は要求にそって修正することもできます。
+
+### モデルの開発と学習
+
+![Feature Engineering](feature_eng.svg)
+
+典型的な TFX パイプラインは [Transform](transform.md) コンポーネントを含みます。
+これは [TensorFlow Transform (TFT)](tft.md) ライブラリを活用して特徴量エンジニアリングを行います。
+Transform コンポーネントは SchemaGen コンポーネントの作成したスキーマを入力とし、[data transformations](https://www.tensorflow.org/tfx/transform/api_docs/python/tft) を適用してモデルの学習に利用する特徴量の作成、組み合わせ、変換を行います。
+欠損値の除去や型の変換についても、それらが推論時のリクエストにも含まれる可能性のある場合、 Transform コンポーネントで実行すべきです。
+TensorFlow のコードを TFX で学習させるよう設計するときには[いくつか重要な検討事項があります。](train.md)
+
+![Modeling and Training](train.svg)
+
+Transform コンポーネントは SavedModel を生成します。
+これは [Trainer](trainer.md) コンポーネントの中で TensorFlow にインポートされ、モデルを記述するコードで利用されます。
+この SavedModel には Transform コンポーネントの作成した、データエンジニアリングで行う変換がすべて含まれています。
+このため、学習時と推論時でまったくおなじコードを用いた同一の変換がなされます。
+Transform コンポーネントで生成された SavedModel を含む、モデルを記述するコードを利用することで、学習用と評価用、両方のデータを利用し、モデルの訓練ができるようになります。
+
+モデルを記述するコードの最後のセクションで、モデルを SavedModel と EvalSavedModel の両方の形式で保存すべきです。 EvalSavelModel 形式で保存するためには Trainer コンポーネントで [TensorFlow Model Analysis (TFMA)](tfma.md) ライブラリをインポートして適用することが必要になるでしょう。
+
+```python
+import tensorflow_model_analysis as tfma
+...
+
+tfma.export.export_eval_savedmodel(
+        estimator=estimator,
+        export_dir_base=eval_model_dir,
+        eval_input_receiver_fn=receiver_fn)
+```
+
+### モデルの振る舞いの分析と理解
+
+![Model Analysis](analysis.svg)
+
+モデルの開発を始め学習させる場合、モデルの振る舞いについて分析し、真に理解することは極めて重要です。
+典型的な TFX パイプラインは [Evaluator](evaluator.md) コンポーネントを含みます。
+このコンポーネントは開発のこのフェーズにおける強力なツールセットを提供する [TensorFlow Model Analysis (TFMA)](tfma.md) ライブラリの能力を活用します。
+Evaluator コンポーネントは保存した EvalSavedModel を入力として受け付けます。また、モデルの振る舞いを可視化し分析する際に利用する `SliceSpecs` のリストを指定できます。
+それぞれの SliceSpec は学習データ中の、カテゴリカルな特徴量における特定のカテゴリ、数値的な特徴量における特定の値域といった、確認したい学習データの切り口を定義します。
+
+たとえば、年間購入金額や地域データ、年齢層、性別といった異なるセグメントの顧客に対するモデルの振る舞いを理解しようと試みることは重要になりえるでしょう。
+特にデータセットがロングテールな分布をしていて、多数派であるグループに対する振る舞いが、小さいけれど重要なグループに対する容認できない振る舞いを覆い隠してしまう場合に、これは特に重要になりえます。
```

Comment:
[weak suggestion]
`特にデータセットが` ==> `データセットが`

This is because `especially` exists only once in the original statement: `This can be especially important for datasets with long tails, where the performance of a dominant group may mask unacceptable performance for important, yet smaller groups.`

Reply:
Thank you for your suggestion, I completely agree with you. "特に" should appear only once.
I made some changes to this sentence In addition to your suggestion to make it fluently.